server/routes/updateBanner.ts

```diff
@@ -0,0 +1,49 @@
+import { Response, NextFunction } from 'express';
+import { TypedRequestBody } from '../types';
+import { DB } from '../database';
+import validateChain from '../util/validateChain';
+import validateRoles from '../util/validateRoles';
+import { CommunityBannerInstance } from '../models/community_banner';
+
+enum UpdateBannerErrors {
+  NoChain = 'Must supply a chain ID',
+  NoPermission = `You do not have permission to update banner`,
+}
+
+
+type UpdateBannerReq = Omit<CommunityBannerInstance, 'id'> & {
+  chain_id: string;
+  banner_text: string;
+};
+
+const updateBanner = async (
+  models: DB,
+  req: TypedRequestBody<UpdateBannerReq>,
+  res: Response,
+  next: NextFunction
+) => {
+  const [chain, error] = await validateChain(models, req.body);
+  if (error) return next(new Error(UpdateBannerErrors.NoChain));
+  const isAdmin = validateRoles(models, req.user, 'admin', chain.id);
+  if (!isAdmin) return next(new Error(UpdateBannerErrors.NoPermission));
```

Comment:
Ideally we use `AppError` and `ServerError` here rather than `next`, like in `viewOffchainVotes`, plus the `success` call to return. The issue with `next` is it always gives a 500 status, which is not correct for invalid args.